### ISSUE TITLE
Issue with mime dependency in svelteesp32@1.6.x with ES Modules

### ISSUE THREAD

**Issue by bcjvandeven** [BCsabaEngine/svelteesp32#35]
Hi,

Thank you for creating and maintaining svelteesp32. It's a great tool for embedding web apps into ESP32 projects!!
However, I encountered an issue with version 1.6.x that might affect others as well.

**Issue:** 
Starting with svelteesp32@1.6.0, the package depends on mime@^4.0.4, which is an ES module. In projects configured with type: "module" (as is standard for modern Node.js and SvelteKit projects), this causes the following error:

`Error [ERR_REQUIRE_ESM]: require() of ES Module .../mime/dist/src/index.js is not supported.`

**Workaround:** 
This is because svelteesp32 still uses require() to load the mime module, which is incompatible with ES modules.
Workaround: I was able to resolve the issue by adding the following override to my package.json:
`"pnpm": {
    "overrides": {
        "mime": "3.0.0"
    }
}
`
This forces pnpm to use a CommonJS-compatible version of mime. However, this is not ideal for all users and might not work for projects using npm or Yarn.

**Suggestions:**
Use dynamic imports: Update svelteesp32 to use import() instead of require() for mime.
Pin mime to version 3.x: This would avoid compatibility issues while still maintaining the current functionality.

**Thanks again for your hard work, and I hope this feedback helps!**

**Comment by BCsabaEngine** [comment]
Thanks for the very detailed description!

I will investigate the situation...

**Comment by BCsabaEngine** [comment]
Do you use it from commandline and referenced as dev dependency? Please clearify the usage details!

**Comment by bcjvandeven** [comment]
Hi 
[index.js.txt](https://github.com/user-attachments/files/18232416/index.js.txt)
BCsabaEngine,

Thank you for maintaining svelteesp32—it's indeed a fantastic tool for embedding web apps into ESP32 projects. I wanted to clarify my usage and provide details regarding the issue I encountered with version 1.6.x.

**Answering Your Questions**
_Do you use it from the command line and reference it as a dev dependency?_
Yes, I use svelteesp32 as a dev dependency and execute it via the command line.

_Why was require("mime") incompatible?_
The issue arises because mime@4.x is an ESM-only module, which cannot be loaded using require(). Projects using type: "module" in package.json (as is standard for SvelteKit and modern Node.js projects) require the use of import().

**Context of the Issue**
I initially used svelteesp32 with a SvelteKit project and Node.js 20.18.1. However, I faced challenges due to the number of generated files, which is a limitation of the ESP32 file system. This led me to migrate to a standalone Svelte project to simplify the structure.

**Issue**
In the standalone Svelte project, with type: "module" configured in package.json (as is standard for modern Node.js and Svelte projects), I encountered the following error:
`Error [ERR_REQUIRE_ESM]: require() of ES Module .../mime/dist/src/index.js is not supported.
`
This error occurred because svelteesp32@1.6.x uses require() to load the mime module, which is incompatible with mime@4.x as it is an ESM-only module.

**Updated Code**
I updated the index.js file of svelteesp32 to replace the use of require("mime") with a dynamic import("mime"), enabling compatibility with ESM and modern Node.js versions (e.g., Node.js 20).

_Compatibility with Modern Node.js:_
The use of import("mime") ensures that the code works seamlessly with mime@4.x and newer Node.js versions (18+), including Node.js 20.
_Dynamic Imports:_
This makes the tool future-proof and compatible with the growing adoption of ESM in the JavaScript ecosystem.
